src/test/java/de/tum/in/www1/artemis/ManagementResourceIntegrationTest.java

```diff
@@ -0,0 +1,110 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.config.audit.AuditEventConverter;
+import de.tum.in.www1.artemis.domain.PersistentAuditEvent;
+import de.tum.in.www1.artemis.repository.PersistenceAuditEventRepository;
+import de.tum.in.www1.artemis.service.AuditEventService;
+import de.tum.in.www1.artemis.service.feature.Feature;
+import de.tum.in.www1.artemis.util.DatabaseUtilService;
+import de.tum.in.www1.artemis.util.RequestUtilService;
+
+public class ManagementResourceIntegrationTest extends AbstractSpringIntegrationTest {
+
+    @Autowired
+    DatabaseUtilService database;
+
+    @Autowired
+    RequestUtilService request;
+
+    @Autowired
+    AuditEventService auditEventService;
+
+    @Autowired
+    PersistenceAuditEventRepository persistenceAuditEventRepository;
+
+    @Autowired
+    AuditEventConverter auditEventConverter;
+
+    private PersistentAuditEvent persAuditEvent;
+
+    @BeforeEach
+    public void initTestCase() {
+        database.addUsers(0, 0, 0);
+        persAuditEvent = new PersistentAuditEvent();
+        persAuditEvent.setPrincipal("student1");
+        persAuditEvent.setAuditEventDate(Instant.now());
+        persAuditEvent.setAuditEventType("type");
+        var data = new HashMap<String, String>();
+        data.put("1", "2");
+        persAuditEvent.setData(data);
+        persistenceAuditEventRepository.deleteAll();
+        persAuditEvent = persistenceAuditEventRepository.save(persAuditEvent);
+
+        var persAuditEvent2 = new PersistentAuditEvent();
+        persAuditEvent2.setPrincipal("student2");
+        persAuditEvent2.setAuditEventDate(Instant.now().minus(5, ChronoUnit.DAYS));
+        persAuditEvent2.setAuditEventType("tt");
+        persAuditEvent2.setData(data);
+        persistenceAuditEventRepository.save(persAuditEvent2);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void toggleFeatures() throws Exception {
+        var features = new HashMap<Feature, Boolean>();
```

Comment:
Please test that at least 5 secured REST calls for the Feature Toggles works when the feature is enabled, and does not work, when the feature is disabled

Reply:
Added 5 calls from different resources